src/decoder.rs

```diff
@@ -538,7 +538,11 @@ impl Decoder {
                     (left16s + top16s - diag16s) as i32,
                 );
             } else {
-                val += get_median(l as i32, t as i32, (l + t - tl) as i32);
+                val += get_median(
+                    l as i32,
+                    t as i32,
+                    (l + t).wrapping_sub(tl) as i32,
```

Comment:
the types can be i32 here and everywhere, we reason about 16bit-per-component at most so far.

Reply:
@lu-zero @Luni-4 would it make sense to get this in for now and leave the overall review open? Without it, things panic in debug builds.

Reply:
Yes, sorry for the wait.